teamup-rocketchat-bot/main.go

```diff
@@ -255,29 +253,104 @@ func checkForMeetings(config *Configuration, chatClient *gorocket.Client) {
 	}
 }
 
+var logOutput string // for initial output
+func init() {
+	// Set the usage for printing usage info
+	flag.Usage = displayHelpMessageWithoutBanner
+	flag.StringVar(&customConfigFile, "config", "", "Point to the configuration (config) file. It overrides the default configuration file located at app directory")
+	flag.StringVar(&customLogPath, "logpath", "", "Set the custom logpath. It overrides the log path specified in configuration (config.yml) file")
+
+	flag.Parse()
+	log.SetPrefix(botPrefix)
+	// If no flags are provided, use defaults
+	fmt.Println(bannerText) // Print the banner, for some  cli awesomeness
+	if len(customConfigFile) == 0 && len(customLogPath) == 0 {
+		customConfigFile = defaultConfigFileName // Set the default config.yml if no parameter was provided for custom config
+		logOutput = "No options specified. Looking for configuration at current directory. Directory for log output will be set at location specified in configuration."
+	} else {
+
+		if len(customConfigFile) == 0 {
+			customConfigFile = defaultConfigFileName
+			logOutput = "Looking for configuration at current directory."
+		} else {
+			logOutput = "Looking for configuration at \"" + customConfigFile + "\"."
+		}
+
+		if len(customLogPath) == 0 {
+			logOutput += " Setting directory specified by configuration for log output."
+		} else {
+			logOutput += " Setting \"" + customLogPath + "\" directory specified by configuration for log output."
+
+		}
+
+	}
+	log.Println(logOutput) // Print for the stdout only
+
+}
 func main() {
 
 	// Setup logger
-	f, err := os.OpenFile(logFile, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0644) // 0644 = user can read and write, other and groups can read
+	// Here, we will use default log path and log file name
+	// And only change to provided input log path after validation
+	var defaultPath string
+	// has CustomPath been used
+	var isCustomPath bool = false
+	if len(customLogPath) == 0 {
+		defaultPath = defaultLogFilePath + "/" + defaultLogFileName
+	} else {
+		defaultPath = path.Clean(customLogPath) + "/" + defaultLogFileName
+		isCustomPath = true
+	}
+	defaultLog, err := os.OpenFile(defaultPath, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0644) // 0644 = user can read and write, other and groups can read
```

Comment:
if there are not enough permissions to create the file the bot crashes ugly
```
TEAMUP-ROCKETCHAT-BOT 2023/01/11 09:05:30 Looking for configuration at "/var/www/rocket.chat/teamup/config.yml". Setting directory specified by configuration for log output.
TEAMUP-ROCKETCHAT-BOT 2023/01/11 09:05:30 open /tmp/teamup-rocket-chat.log: permission denied
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x529229]

goroutine 1 [running]:
log.(*Logger).Output(0x0, 0x1?, {0xc000128200, 0x33})
	/usr/local/go/src/log/log.go:179 +0x89
log.(*Logger).Fatal(0xc0001281c0?, {0xc000125f00?, 0x441?, 0x7c41f0?})
	/usr/local/go/src/log/log.go:231 +0x45
main.main()
	/tmp/hubot/teamup-rocketchat-bot/main.go:307 +0x158
```

also I would suggest to make the permissions 0640, others (anyone on the system) does not need to be able to read the logfile

Reply:
I have  made an issue at [https://github.com/JankariTech/hubot/issues/15](https://github.com/JankariTech/hubot/issues/15) . Let's not add more features in this PR.